### DISCUSSION TITLE
Why does `strat.cpp` set floors in `UserWorkInLoop()`?

### DISCUSSION THREAD

**Discussion by msbc** [PrincetonUniversity/athena#611] [General]
Why does the `strat.cpp` pgen set energy floors in `UserWorkInLoop()`? Shouldn't the floors already be taken care of by the flooring functions?

**Comment by tomidakn** [comment]
I guess there is no significant reason but it is simply ported from Athena 4.2. I checked Athena 4.2 and it had a density (but no pressure) floor in the Userwork_in_loop function.

**Reply by msbc** [reply]
Okay, thanks. That makes sense. I was modifying the pgen to work for a general EOS, and I just wanted to check that I can safely remove that floor.

**Reply by felker** [reply]
I would advise removing any unnecessary floors from `master`
